kanidm_book/src/examples/k8s_ingress_example.md

```diff
@@ -0,0 +1,279 @@
+# Kubernetes Ingress 
+
+Guard your Kubernetes ingress with Kanidm authentication and authorization.
+
+## Prerequisites
+
+The following are used in this example:
+
+- [Kanidm](../installing_the_server.html)
+- [Kubernetes v1.23 or above](https://docs.k0sproject.io/v1.23.6+k0s.2/install/)
+- [Nginx Ingress](https://kubernetes.github.io/ingress-nginx/deploy/)
+- [A Domain Name](https://domains.google/)
+- [CertManager - With Cluster Issuer](https://cert-manager.io/docs/installation/)
+- [ShellCheck](https://github.com/koalaman/shellcheck#user-content-installing)
+
+You can set up your certificates manually instead of using CertManager if you are comfortable doing so. If it is your first time setting up CertManager, take your time with installation and validation. For the `http01` solver the ingress port (8089) needs to be accessible from the internet - DNS validation is recommended.
+
+## Instructions
+
+1. Create a script `deploy.sh` from the content below, replacing every `<string>` (drop the `<>`) with appropriate values, then run it. This will create the user, group and OAUTH2 resource.
+
+```shell
+#!/bin/sh
+
+# Analysis self for common errors.
+shellcheck ./*.sh || exit 1
+
+# Login
+echo "User: idm_admin"
+kanidm login --name idm_admin
+
+# User Setup
+kanidm account create k8s_example_user "K8s Example User" --name idm_admin
+echo "Updating the credentials for k8s_example_user..."
+kanidm account credential update k8s_example_user --name idm_admin
+# > password
+# > commit
+kanidm account person extend k8s_example_user --legalname "John K8s Doe" --mail "jkd@email.address" --name idm_admin
+
+# Create Group
+kanidm group create k8s_example_group --name idm_admin
+
+# Attach User to Group
+kanidm group add_members k8s_example_group k8s_example_user --name idm_admin
+
+# Setup Oauth Resource
+displayname="The Example"
+name=<oauth2_rs_name>
+domain=<host>
+echo "User: admin"
+kanidm login --name admin
```

Comment:
I don't really like the CLI being in "scripts" since we aren't committing to the CLI layout as a stable interface. So at the very least, we could warn about this, or segment it so that we can show which actions are "manual" to run?

Reply:
After a bit more thinking I think a lot of the "account setup" and oauth steps can be directed to the relevant book chapters instead, rather than being part of the script. Documentation should be describing "how to achieve something" rather than a "copy-paste" script recipe.

Reply:
I will make the Kanidm portions tutorial-like.